src/ui/public/stringify/types/_numeral.js

```diff
@@ -1,59 +1,59 @@
 import _ from 'lodash';
+import numeral from 'numeral';
 import 'ui/field_format_editor/numeral/numeral';
-import { IndexPatternsFieldFormatProvider } from 'ui/index_patterns/_field_format/field_format';
-import { BoundToConfigObjProvider } from 'ui/bound_to_config_obj';
+import { FieldFormat } from 'ui/index_patterns/_field_format/field_format';
 
-export function StringifyTypesNumeralProvider(Private) {
-  const FieldFormat = Private(IndexPatternsFieldFormatProvider);
-  const BoundToConfigObj = Private(BoundToConfigObjProvider);
-  const numeral = require('numeral')();
+const numeralInst = numeral();
 
-  _.class(Numeral).inherits(FieldFormat);
-  function Numeral(params) {
-    Numeral.Super.call(this, params);
-  }
-
-  Numeral.prototype._convert = function (val) {
-    if (val === -Infinity) return '-∞';
-    if (val === +Infinity) return '+∞';
-    if (typeof val !== 'number') {
-      val = parseFloat(val);
-    }
+_.class(Numeral).inherits(FieldFormat);
+export function Numeral(params) {
+  Numeral.Super.call(this, params);
+}
 
-    if (isNaN(val)) return '';
+Numeral.prototype._convert = function (val) {
+  if (val === -Infinity) return '-∞';
+  if (val === +Infinity) return '+∞';
+  if (typeof val !== 'number') {
+    val = parseFloat(val);
+  }
 
-    return numeral.set(val).format(this.param('pattern'));
-  };
+  if (isNaN(val)) return '';
 
+  return numeralInst.set(val).format(this.param('pattern'));
+};
 
-  Numeral.factory = function (opts) {
-    _.class(Class).inherits(Numeral);
-    function Class(params) {
-      Class.Super.call(this, params);
-    }
 
-    Class.id = opts.id;
-    Class.title = opts.title;
-    Class.fieldType = 'number';
+Numeral.factory = function (opts) {
+  _.class(Class).inherits(Numeral);
+  function Class(params, getConfig) {
+    Class.Super.call(this, params);
 
-    Class.paramDefaults = opts.paramDefaults || new BoundToConfigObj({
-      pattern: '=format:' + opts.id + ':defaultPattern',
-    });
+    this.getConfig = getConfig;
+  }
 
-    Class.editor = {
-      template: opts.editorTemplate || require('ui/field_format_editor/numeral/numeral.html'),
-      controllerAs: 'cntrl',
-      controller: opts.controller || function () {
-        this.sampleInputs = opts.sampleInputs;
-      }
-    };
+  Class.id = opts.id;
+  Class.title = opts.title;
+  Class.fieldType = 'number';
 
-    if (opts.prototype) {
-      _.assign(Class.prototype, opts.prototype);
+  Class.editor = {
+    template: opts.editorTemplate || require('ui/field_format_editor/numeral/numeral.html'),
+    controllerAs: 'cntrl',
+    controller: opts.controller || function () {
+      this.sampleInputs = opts.sampleInputs;
     }
+  };
 
-    return Class;
+  Class.prototype.getParamDefaults = function () {
+    const paramDefaults = _.get(opts, 'paramDefaults', {});
```

Comment:
I'm not a huge fan of splitting the param defaults up like this. What about something like this?

```js
Class.prototype.getParamDefaults = function () {
  if (_.has(opts, 'getParamDefaults')) {
    return opts.getParamDefaults(this.getConfig)
  }

  return {
    pattern: this.getConfig(`format:${opts.id}:defaultPattern`)
  }
}
```

Then the instances can specify the `getParamDefaults(getConfig)` option and read config themselves.